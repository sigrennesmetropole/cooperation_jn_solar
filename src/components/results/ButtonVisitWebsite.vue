<script setup lang="ts">
import goToWhite from '@/assets/icons/icon-external-link-white-thin.svg'

const props = defineProps<{
  link: string
}>()
</script>

<template>
  <a
    :href="props.link"
    target="_blank"
    class="bg-black shadow-sm rounded-lg gap-3 px-4 py-3 w-fit flex flex-row items-center justify-center mt-4"
  >
    <img class="w-5 h-5" :src="goToWhite" alt="" />
    <span class="font-dm-sans text-white text-base font-bold">
      Visiter le site
    </span>
  </a>
</template>
