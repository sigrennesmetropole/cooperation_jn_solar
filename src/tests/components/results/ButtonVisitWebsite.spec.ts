import { mount } from '@vue/test-utils'
import ButtonVisitWebsite from '@/components/results/ButtonVisitWebsite.vue'

describe('ButtonVisitWebsite.vue', () => {
  const link = 'https://www.example.com/'

  it('renders the component correctly', () => {
    const wrapper = mount(ButtonVisitWebsite, {
      props: { link },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders a link with the correct href', () => {
    const wrapper = mount(ButtonVisitWebsite, {
      props: { link },
    })
    const a = wrapper.find('a')
    expect(a.attributes('href')).toBe(link)
  })

  it('has the correct target attribute on the link', () => {
    const wrapper = mount(ButtonVisitWebsite, {
      props: { link },
    })
    const a = wrapper.find('a')
    expect(a.attributes('target')).toBe('_blank')
  })

  it('has the correct link text', () => {
    const wrapper = mount(ButtonVisitWebsite, {
      props: { link },
    })
    const span = wrapper.find('span')
    expect(span.text()).toBe('Visiter le site')
  })

  it('renders the correct img alt attribute', () => {
    const wrapper = mount(ButtonVisitWebsite, {
      props: { link },
    })
    const img = wrapper.find('img')
    expect(img.attributes('alt')).toBe('')
  })

  it('renders the correct img src attribute', () => {
    const wrapper = mount(ButtonVisitWebsite, {
      props: { link },
    })
    const img = wrapper.find('img')
    expect(img.attributes('src')).toContain('icon-external-link-white-thin.svg')
  })
})
