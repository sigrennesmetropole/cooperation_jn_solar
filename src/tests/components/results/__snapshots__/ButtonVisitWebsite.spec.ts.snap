// Vitest Snapshot v1

exports[`ButtonVisitWebsite.vue > renders the component correctly 1`] = `"<a href=\\"https://www.example.com/\\" target=\\"_blank\\" class=\\"bg-black shadow-sm rounded-lg gap-3 px-4 py-3 w-fit flex flex-row items-center justify-center mt-4\\"><img class=\\"w-5 h-5\\" src=\\"/src/assets/icons/icon-external-link-white-thin.svg\\" alt=\\"\\"><span class=\\"font-dm-sans text-white text-base font-bold\\"> Visiter le site </span></a>"`;
